File: test-suite/golden-tests/defunc-record-closures-rust-py/consumer.py
def use_ops(ops, x):
    return ops["inc"](x) + ops["scale"](x)


def use_list(fs, x):
    return fs[0](x) + fs[1](x)


def use_tuple(t, x):
    return t[0](x) + t[1](x)


def use_opt(f, x):
    return x if f is None else f(x)

File: test-suite/golden-tests/defunc-record-closures-pr/consumer.R
use_ops <- function(ops, x) {
  ops[["inc"]](x) + ops[["scale"]](x)
}

use_list <- function(fs, x) {
  fs[[1]](x) + fs[[2]](x)
}

use_tuple <- function(t, x) {
  t[[1]](x) + t[[2]](x)
}

use_opt <- function(f, x) {
  if (is.null(f)) x else f(x)
}
